### PR TITLE
feat(snowflake): add support for `imports` and `packages` UDF configuration arguments

### PR DESCRIPTION
**Pull request by cpcloud**
Adds support for passing `imports` and `packages` to snowflake UDFs per [this](https://ibis-project.zulipchat.com/#narrow/stream/405329-snowflake/topic/Scalar.20UDFs) Zulip topic.

### CODE REVIEW

File: ibis/backends/snowflake/tests/test_udf.py [hunk 1]
```diff
@@ -92,3 +92,67 @@ def test_builtin_agg_udf(con):
...
+
+        import joblib
+        import pandas as pd
+
+        import_dir = sys._xoptions.get("snowflake_import_directory")
+        model = joblib.load(f"{import_dir}model.joblib")
+        df = pd.concat(
+            [carat_scaled, cut_encoded, color_encoded, clarity_encoded], axis=1
+        )
+        df.columns = ["CARAT_SCALED", "CUT_ENCODED", "COLOR_ENCODED", "CLARITY_ENCODED"]
```
Comment (cpcloud) [inline ibis/backends/snowflake/tests/test_udf.py:117]: Not a huge fan of having to do this renaming, but I'm not entirely sure how it can be avoided.

**Comment by sfc-gh-twhite** [comment]
I just tested this out, and it works! 

I was fiddling around with trying to do the imports outside of the function code, and that doesn't work, so that's a bummer, but I'm not sure how we could get around it, the Snowpark UDF registration automatically makes the reference.

**Comment by cpcloud** [comment]
@sfc-gh-twhite It looks like there's quite a bit of machinery in Snowpark to do this, including pickling the function and uploading code that then inlines the pickled bytes and deserializes those bytes into a callable. Pickling has a greater chance of capturing imports which would explain why snowpark can do that and ibis can't.

We'll probably have to go down this path eventually.

How important is this for your workflow?

**Comment by sfc-gh-twhite** [comment]
It works great as it is right now. I think having the code picked up and the modules made available without being specified within the function code is more of a "nice to have" feature that could be added later. This helps avoid going back and forth between Ibis and Snowpark.

**Comment by cpcloud** [comment]
@gforsyth Thanks for pointing that out. I need to figure out what to do about the model. I think I'll throw it in a gcs bucket and upload it to a stage in CI.

**Comment by cpcloud** [comment]
Clouds are passing:

```
cloud in 🌐 falcon in …/ibis on  snowflake-udf-args is 📦 v7.1.0 via 🐍 v3.10.13 via ❄️  impure (ibis-3.10.13-env)
❯ pytest -m 'bigquery or snowflake' -n 8 --dist loadgroup --snapshot-update
========================================================================================== test session starts ==========================================================================================
platform linux -- Python 3.10.13, pytest-7.4.3, pluggy-1.3.0
benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Using --randomly-seed=640021664
rootdir: /home/cloud/ibis
configfile: pyproject.toml
plugins: benchmark-4.0.0, clarity-1.0.1, mock-3.12.0, randomly-3.15.0, cov-4.1.0, repeat-0.9.3, hypothesis-6.91.0, httpserver-1.0.8, xdist-3.5.0, snapshot-0.9.0
8 workers [3076 items]
xxxxx.xxxxxx.xxx.ssssssssssssssssssssss.xx...xxxxxxx.xxxxx..xsxx..x.....xx..xxxx...xxxxx..............x...x..x..x..x.x..xx...x.xx..xx..x..x.xx...xxx....x...x......x.x.xxxx.....xxx....xx.....x.. [  6%]
..x..x....x..x.xx..xxx....x.x.xx.....x..xx...xx...x..x...........xxx..xxx.x......x..x..x....xx......x.x......x..x..x...x....x.........x......x.x................................................. [ 12%]
..........................x............................................xxx..x.xx..x...xxxxx...x..xxxxxxx.x.xxxxxxxxxxxxxxxxxx.xx.x.x..x.......xx..............s....x.xs.......s......x..x..x....x [ 18%]
...........x..x.x.......xx................x.xx......xx.....x.....xx............x......x..x.........x......x.......x...x......x.xx.x...x.......x.........x....xx..........x.xx.....xx..x...x.x.... [ 25%]
x....x............x..............x.......x...x......x....x...x..x....x.x...xx...........x..x..x....x....x.........x..............x..x...x......xx..........x..................................... [ 31%]
....................................................x.xx.xxxx.x..x.................................x.......x.....................x...x...x.............x.x...x......x....xx..x..x....x....x.x.x.. [ 37%]
......xx.....x.....x......x.............x......x........x..xx...........x......x........x.x..x..x..xx.....xxx.......x...x.x.........................x.........x................s......x.........x [ 43%]
......x.x.x....x.xxx..x.x..............x.xx......x.x.............x.x................x.........x...x..............x.x.x.xx...x......x...x.x....x...................xx.................x.......x... [ 50%]
...............x.................x...........x................x.......x..............x..x...xxx.x.x.xxxxxxx..xx..x......x.......x......................................x..........x.............. [ 56%]
.......x.x.................................x.............x.x.........x..........x..x......x..........................x.......x............x..........x.....x...................x..x..x........... [ 62%]
.......x......x..........x....x...x.........x............x..........x.......x.x........x........x.x....x..x.x........x....x.......x........x...x....x.....x...............xx.x................... [ 69%]
......................................................................x..x.xxx............................................x................................................x.....x....x.......... [ 75%]
.....x.x......s.x..x..x..xxx.x.x..x......x...x.....xx...x.x....x.......x........................................x..................x..........xx..................x..........................x... [ 81%]
....x..x............xx...............x.........x.x..xx..xxx.........x.x.............x.........x........s......x.................................................................................. [ 87%]
...................................................................................................................s............................................................................. [ 94%]
..........................................................x...............................................x.................................................................x........             [100%]
======================================================================= 2601 passed, 30 skipped, 445 xfailed in 668.72s (0:11:08) =======================================================================
```

**Review by gforsyth — APPROVED** [review]
:shipit:

**Pull request merged**